avocado/core/nrunner.py

```diff
@@ -229,13 +230,13 @@ class ExecTestRunner(ExecRunner):
     Runnable attributes usage is identical to :class:`ExecRunner`
     """
     def run(self):
-        for status in super(ExecTestRunner, self).run():
-            if 'returncode' in status:
-                if status['returncode'] == 0:
-                    status['status'] = 'pass'
+        for output in super(ExecTestRunner, self).run():
```

Comment:
The variable name here is tricky indeed (as naming usually is). I feel that naming this `most_current_execution_state` would be too verbose, but, would be very accurate.

We've been plagued by not having descriptive enough terms, such as `runner` being used in more than on place, so even though this is very verbose, I think it'd be a good idea.